### ISSUE TITLE
Troubleshooting our whitening transform

### ISSUE THREAD

**Issue by gwaygenomics** [cytomining/pycytominer#90]
Whitening is giving us a problem in at least three different places.

1. @jccaicedo is using it in DeepProfiler (see [here](https://github.com/broadinstitute/DeepProfilerExperiments/blob/35d53cf9bf83676b150696d1b2a1b64db618028d/ta-orf/profiling.py))
2. @niranjchandrasekaran is using an implementation of `ZCA` whitening in the JUMP CP project (a private repo)
    * Implementation: https://github.com/mwv/zca
3. I am having trouble with it in the LINCS Cell Painting project (see broadinstitute/lincs-cell-painting#38)
    * I am using the pycytominer implementation [here](https://github.com/cytomining/pycytominer/blob/master/pycytominer/cyto_utils/transform.py#L13).
    * Documented in #89

I propose that we define a common implementation, and that we use this issue to discuss. Even if everyone doesn't agree on an implementation, it would be at least good to get a working version on pycytominer.

**Comment by gwaygenomics** [comment]
After writing this, perhaps it is just me that has the problem! 😆 Either way, the implementations for all three projects above are different.

**Comment by niranjchandrasekaran** [comment]
Sounds good! 😄  

Based on the little bit of time that I have spent exploring the implementation of whitening in the context of morphological profiles, these are the unresolved questions that I can think of (some are better resolved than others)

1. Which flavor of whitening to use - `PCA`, `ZCA` or others?
2. Which matrix to use - `covariance matrix` vs. `correlation matrix`?
3. Should whitening be applied to 
    a. each plate individually
    b. each platemap (replicate plates)
    c. all plates 
    d. some combination of the above 

Based on my experience with JUMP-CP the following are my answers to these questions though I am unsure if everyone's answer will be same as mine. 

1. `ZCA`
2. `correlation matrix`
3. each plate individually

Apart from these, IIUC, the implementation of whitening could also be different depending on whether it being applied to single cell profiles or to perturbation level profiles.

**Comment by gwaygenomics** [comment]
This is very helpful @niranjchandrasekaran - thanks! A couple of comments and followup questions:

* I think we'll want to implement both `PCA` and `ZCA` whitening in pycytominer and provide the option for users to select either
  * Given the answer that `ZCA` seems to perform the best, we should make this default
* We'll also provide the `covariance` vs. `correlation` option to a user
  * Although I have never seen a whitening implementation for correlation matrices... I think we should stick with `covariance` as the default here, since this is what most people will expect. @niranjchandrasekaran - happy to change my mind here if you've come across other evidence (also, I assume we're talking pearson correlation matrix?)
* What plate to apply whitening is beyond the scope of pycytominer, but is certainly a very important question.
* On a similar note, @niranjchandrasekaran - at what aggregate profile level do you typically perform whitening? Is it performed before or after z-score normalization?

**Comment by niranjchandrasekaran** [comment]
> happy to change my mind here if you've come across other evidence (also, I assume we're talking pearson correlation matrix?)

Yes, I used Pearson correlation for my ZCA whitening. Regarding the use of `correlation` matrix, I recently came across [this paper](https://arxiv.org/pdf/1512.00809.pdf) where they compared several whitening methods and found that the performance of both `PCA` and `ZCA` whitening improved with the use of `correlation` matrix. Other than this, I haven't come across any other evidence, though during one of our check ins, @shntnu was surprised to hear that `correlation` matrix was not used by default during whitening. Perhaps @shntnu has more insights about this.

>  Is it performed before or after z-score normalization?

Since I was comparing the performance of different normalization methods, I applied it instead of z-score normalization. This might also explain why my implementation of ZCA required the use `correlation` matrix because without z-score normalization done beforehand, the PCs were dominated by features with the largest values.
